operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md

```diff
@@ -0,0 +1,173 @@
+# Create a Session Template to Boot Compute Nodes with Scalable Boot Projection Service (SBPS)
+
+When [compute nodes](../../glossary.md#compute-node-cn) are booted, the [Scalable Boot Projection Service (SBPS)](../../glossary.md#scalable-boot-projection-service-sbps)
+projects the root file system \(`rootfs`\) over the network to the compute nodes.
+
+
+This page covers the appropriate contents for a BOS session template in order to use SBPS.
+
+- [Boot set S3 parameters](#boot-set-s3-parameters)
+- [Boot set `rootfs_provider` parameters](#boot-set-rootfs_provider-parameters)
+  - [`<transport>`](#transport)
+  - [`<schema version>`](#schema-version)
+  - [`<IQN Domain>`](#iqn-domain)
+  - [`<DNS SRV record reference>`](#dns-service-srv-record-reference)
+  - [`<client discovery timeout in seconds>`](#client-discovery-timeout-in-seconds)
+  - [Example `rootfs_provider_passthrough`](#example-rootfs_provider_passthrough)
+- [`root=` kernel parameter](#root-kernel-parameter)
+- [Example session template input file](#example-session-template-input-file)
+- [Creating a BOS session using the new template](#creating-a-bos-session-using-the-new-template)
+
+## Boot set S3 parameters
+
+The session template boot set contains several [Simple Storage Service (S3)](../../glossary.md#simple-storage-service-s3) parameters.
+These are listed below, along with the appropriate values to use.
+
+- `type`: Set to `s3`
+- `path`: Set to `s3://<BUCKET_NAME>/<KEY_NAME>`
+  - `<BUCKET_NAME>` is set to `boot-images`
+  - `<KEY_NAME>` is set to the image ID that the [Image Management Service (IMS)](../../glossary.md#image-management-service-ims) created when it generated the boot artifacts.
+- `etag`: Set to `<etag\>`
+
+## Boot set `rootfs_provider` parameters
+
+The Scalable Boot Projection Service (SBPS) is an optional provider for the `rootfs` on compute nodes.
+
+For SBPS, the `rootfs_provider_passthrough` boot set parameter is customized according to the following format:
+
+```text
+rootfs_provider_passthrough=<transport>:<schema version>:<IQN Domain>:<DNS SRV record reference>:<client discovery timeout in seconds>
+```
+
+The following values need to be set in the boot set of the session template in order to make SBPS the `rootfs` provider:
+
+- `"rootfs_provider":` Set to `"sbps"`
+- To use the Node Management Network (NMN) for content projection,
+  - `"rootfs_provide_passthrough"`: Set to `"sbps:v1:iqn.2023-06.csm.iscsi:_sbps-nmn._tcp.local:300"`
+- To use the High Speed Network (HSN) for content projection,
+  - `"rootfs_provide_passthrough"`: Set to `"sbps:v1:iqn.2023-06.csm.iscsi:_sbps-hsn._tcp.local:300"`
+
+Note that `iqn.2023-06.csm.iscsi` is an example of the IQN domain. Provide an IQN domain specific to the customer.
```

Comment:
This will be a fixed value for now.

Reply:
@jeremy-duckworth , are you saying that the fixed value _is_ 'iqn.2023-06.csm.iscsi' or just that there will be a fixed value for the domain name? 
In this documentation, I am trying to give the sysadmins a clear understanding of what things are, but also of what they can and cannot just cut-and-paste into a BOS session template.

Reply:
The fixed value is `iqn.2023-06.csm.iscsi`, the worker nodes will show up in the style of `iqn.2023-06.csm.iscsi:ncn-w005`. I believe COS is using the xname for the initiator IDs (same IQN), but Kazuya can confirm.

Reply:
Okay, @jeremy-duckworth , queuing up the dumb questions.
1. Will the string `iqn.2023-06.csm.iscsi` be a fixed string that _every_ site uses for its domain name?
Or will it be site-specific like `iqn.hi-im-nersc.csm.iscsi` or `iqn.hi-im-lanl.csm.iscsi`?

I think you were alluding to this in your first comment 'This domain name will be based on the system domain name.'



2. In your above example, you replaced `_sbps-hsn._tcp.local` with `ncn-w005`.  Was that your intent? 
'ncn-w005' contains none of the DNS SRV attributes except the symbolic name. It doesn't have the transport protocol or domain name.
* `_sbps-hsn` is the symbolic name of the service
* `_tcp` is the transport protocol
* `local` is the domain name for which the record is valid

Reply:
# 1 - fixed string. This should not be conflated with anything DNS related. 

# 2 - `ncn-w005` is the endpoint/node designation that forms a fully qualified iSCSI IQN.

Reply:
I was conflating it with DNS and the system domain name. I'm still unclear on the IQN domain versus a DNS system domain.